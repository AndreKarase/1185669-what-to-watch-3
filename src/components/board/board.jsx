import React, {PureComponent, Fragment} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer.js';
import MoviesList from '../movies-list/movies-list.jsx';
import GenreList from '../genre-list/genre-list.jsx';
import ShowMore from '../show-more/show-more.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.js';

const GenreListWrapped = withActiveItem(GenreList);
const MoviesListWrapped = withActiveItem(MoviesList);

class Board extends PureComponent {
  render() {
    const {maxMoviesCount, movies, moviesAll, onChangeActiveGenre, onShowMoreClick, resetMaxMoviesCount} = this.props;

    return (
      <Fragment>
        <GenreListWrapped
          movies={moviesAll}
          activeItem={`All movies`}
          onChangeActiveGenre={(genre) => {
            onChangeActiveGenre(genre);
            resetMaxMoviesCount();
          }}
        />
        <MoviesListWrapped
          movies={movies}
          activeItem={movies[0]}
          maxCount={maxMoviesCount}
        />
        {maxMoviesCount < movies.length ?
          <ShowMore
            onClick={onShowMoreClick}
          /> : null
        }
      </Fragment>
    );
  }
}

Board.propTypes = {
  maxMoviesCount: PropTypes.number.isRequired,
  movies: PropTypes.array.isRequired,
  moviesAll: PropTypes.array.isRequired,
  onChangeActiveGenre: PropTypes.func.isRequired,
  onShowMoreClick: PropTypes.func.isRequired,
  resetMaxMoviesCount: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  moviesAll: state.moviesAll,
  movies: state.movies,
  activeGenre: state.activeGenre
});

const mapDispatchToProps = (dispatch) => ({
  onChangeActiveGenre(genre) {
    dispatch(ActionCreator.changeActiveGenre(genre));
    dispatch(ActionCreator.getMovies(genre));
  }
});

export {Board};
export default connect(mapStateToProps, mapDispatchToProps)(Board);

