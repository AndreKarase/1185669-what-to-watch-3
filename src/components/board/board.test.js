import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {Board} from './board.jsx';

const mockStore = configureStore([]);

const movies = [{
  title: `1`,
  previewImage: ``,
  posterImage: ``,
  backgroundImage: ``,
  genre: ``,
  releaseDate: 1998,
  preview: ``,
  description: ``,
  rating: 10,
  scoresCount: 1,
  director: ``,
  starring: [``, ``, ``],
}, {
  title: `2`,
  previewImage: ``,
  posterImage: ``,
  backgroundImage: ``,
  genre: ``,
  releaseDate: 1998,
  preview: ``,
  description: ``,
  rating: 10,
  scoresCount: 1,
  director: ``,
  starring: [``, ``, ``],
}];

it(`Should render Board`, () => {
  const store = mockStore({
    activeGenre: `All genres`,
    moviesAll: movies,
    movies
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Board
          maxMoviesCount={1}
          movies={movies}
          moviesAll={movies}
          onChangeActiveGenre={() => {}}
          onShowMoreClick={() => {}}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
