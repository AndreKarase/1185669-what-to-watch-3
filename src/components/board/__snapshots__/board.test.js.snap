// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Board 1`] = `
Array [
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        
      </a>
    </li>
  </ul>,
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="1"
          height="175"
          src=""
          width="280"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
          >
            1
          </a>
        </h3>
      </div>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>,
]
`;
